mypy/checkmember.py

```diff
@@ -371,8 +371,27 @@ def validate_super_call(node: FuncBase, mx: MemberContext) -> None:
 def analyze_type_callable_member_access(name: str, typ: FunctionLike, mx: MemberContext) -> Type:
     # Class attribute.
     # TODO super?
-    ret_type = typ.items[0].ret_type
+    item = typ.items[0]
+    ret_type = item.ret_type
     assert isinstance(ret_type, ProperType)
+
+    # The following is a hack. mypy often represents types as CallableType, where the signature of
+    # CallableType is determined by __new__ or __init__ of the type (this logic is in
+    # type_object_type). Then if we ever need the TypeInfo or an instance of the type, we fish
+    # around for the return type in CallableType.type_object. Unfortunately, this is incorrect if
```

Comment:
> we fish around for the return type in CallableType.type_object

Where do we do this? This function certainly didn't, going for simple `.ret_type`

Reply:
I clarified the comment a little bit. `type_object` also just uses `ret_type`. Also, thanks for the suggestion to use bound_args, I added you as a co-author of this PR!